test/unit/test_runner_config.py

```diff
@@ -189,9 +181,16 @@ def test_generate_ansible_command():
     cmd = rc.generate_ansible_command()
     assert cmd == ['ansible-playbook', '-i', '/inventory', 'main.yaml']
 
-    rc.extra_vars = '/env/extravars'
+    rc.extra_vars = dict(test="key")
     cmd = rc.generate_ansible_command()
-    assert cmd == ['ansible-playbook', '-i', '/inventory', '-e', '@/env/extravars', 'main.yaml']
+    assert cmd == ['ansible-playbook', '-i', '/inventory', '-e', '\'test="key"\'', 'main.yaml']
+
+    with patch.object(rc.loader, 'isfile', side_effect=lambda x: True):
+        cmd = rc.generate_ansible_command()
+        assert cmd == ['ansible-playbook', '-i', '/inventory', '-e', '@/env/extravars', '-e', '\'test="key"\'', 'main.yaml']
```

Comment:
I like the use of `@file_name` for passing extra vars to ansible.   It might be nice to make another temporary file for passing the API provided extra vars too.

Reply:
Is there a limit on the number of arguments that can be passed to ansible?  That might limit the number of extra vars that can be passed via the API.

Reply:
When the user provides `self.extra_vars` we treat it as one big blob, but yeah... we could run into problems there. The idea if that's the case would be for the user to put that information *into* `env/extravars`

Reply:
That makes sense.